### ISSUE TITLE
Thermistor

### ISSUE THREAD

**Issue by Jaaannis** [bigtreetech/SKR-Pico#3]
Hi!
is there a PIN that i can use for thermistor? (3950 eg)
Can´t figure that out.
Thanks :-)

**Comment by Pieterv24** [comment]
for thermistor you need a pin that has access to the ADC.
These are gpio26 to gpio29 according to the docs of the RP2040 chip.
And since gpio26 and gpio27 are already used for the hotend, and bed thermistors.
gpio28 is for some reason part of the pins driving the z stepper driver (unaccesible).
That only leaves pin gpio29.

So if you already have a servo/bltouch etc connected to that port, you might need to change its location.
Possibly use gpio20 for that if it's still free

**Comment by bistory** [comment]
> for thermistor you need a pin that has access to the ADC. These are gpio26 to gpio29 according to the docs of the RP2040 chip. And since gpio26 and gpio27 are already used for the hotend, and bed thermistors. gpio28 is for some reason part of the pins driving the z stepper driver (unaccesible). That only leaves pin gpio29.
> 
> So if you already have a servo/bltouch etc connected to that port, you might need to change its location. Possibly use gpio20 for that if it's still free

Unfortunately, it doesn't work as expected : with a Beta 3950, I get crazy values like 330°C when ambiant temperature is 20°C...

**Comment by justgook** [comment]
same here

**Comment by xp4ndeR** [comment]
For the thermistor TH0 and THB you have additional circuitry according to the documentation with resistors :

![image](https://user-images.githubusercontent.com/83035188/224828512-90dd8fa3-6548-492a-b052-eff5c89b3be5.png)

Personally I would try to edit my printer config in klipper and play with "pullup_resistor" and set it to 0 ( or even define a custom thermistor to account for the different resistance.

If not on klipper, I would solder a 4.7Kohm resistance inline with my sensor and see if it's any better. 
 
https://www.klipper3d.org/Config_Reference.html#thermistor

**Comment by justgook** [comment]
I think I need to retest it with other resters that really works, and if they fail too, then can think about adding additional circuitry

**Comment by justgook** [comment]
checked - still get same 340-360C - so - to make it work based on circuitry I need connect 4k7 resistor between positive, and gpio29, and then add in series with thermistor? (sorry not very good in reading schematics)

btw I need connect trimester between ground and gpio29, I'm right?

**Comment by xp4ndeR** [comment]
I'm better at reading the documentation than schematics tbh :D

My understanding (which is limited) is that the additional circuit is meant to provide a more stable power source (ADC_VREF)  or reduce power draw from the RPi2040 and probably provide better reading quality of the thermistor and also security with the diodes. But I'm kinda guessing as the electronics lesson were a long time ago !

But I think connecting a 4k7 resistor in series with your thermistor is enough to make it work. I shouldn't matter which side as resistor add value up ( thermistor is just a resistor that change depending of temp)

Worst case you have the wrong value again :D

**Comment by mysy200** [comment]
Yes, SKR Pico really need provide other pin for thermistor measure. like I use V0-Umbilical have other 100K  chamber temperature.
Any one know it?
